src/services/infra/InfraService.ts

```diff
@@ -91,4 +105,174 @@ export default class InfraService {
+       */
+
+      const dnsInfo = await this.launchesService.getDomainAssociationRecord(
+        primaryDomain,
+        appId
+      )
+
+      const certificationRecord = this.parseDNSRecords(
+        dnsInfo.domainAssociation?.certificateVerificationDNSRecord
+      )
+      if (!certificationRecord) {
+        throw new Error(`error while parsing ${dnsInfo}`)
+      }
+
+      const {
+        source: domainValidationSource,
+        target: domainValidationTarget,
+      } = certificationRecord
+
+      const subDomainList = dnsInfo.domainAssociation?.subDomains
+      if (!subDomainList || !subDomainList[0].dnsRecord) {
+        throw Error("subdomain list not created yet")
+      }
+
+      const primaryDomainInfo = this.parseDNSRecords(subDomainList[0].dnsRecord)
+
+      if (!primaryDomainInfo) {
+        throw Error("primary domain info not created yet")
+      }
+
+      /**
+       * shape of dnsInfo.domainAssociation.subDomains:
+       * {
+       *   dnsRecord: "CNAME gibberish.cloudfront.net",
+       *   subDomainSettings: {
+       *     branchName : "master",
+       *     prefix? : "www"
+       *   }
+       * }
+       */
+
+      const primaryDomainTarget = primaryDomainInfo.target
+      const redirectionDomainList = dnsInfo.domainAssociation?.subDomains?.filter(
+        (subDomain) => subDomain.subDomainSetting?.prefix
+      )
+
+      /**
+       * Amplify only stores the prefix.
+       * ie: if I wanted to have a www.blah.gov.sg -> giberish.cloudfront.net,
+       * amplify will store the prefix as "www". To get the entire redirectionDomainSource,
+       * I would have to add the prefix ("www") with the primary domain (blah.gov.sg)
+       */
+      const redirectionDomainSource = `${redirectionDomainList?.[0].subDomainSetting?.prefix}.${primaryDomain}`
+
+      const userId = agency.id
+      const newLaunchParams = {
+        userId,
+        siteId,
+        primaryDomainSource: primaryDomain,
+        primaryDomainTarget,
+        domainValidationSource,
+        domainValidationTarget,
+        redirectionDomainSource,
+      }
+
+      // Create launches records table
+      const launchesRecord = await this.launchesService.create(newLaunchParams)
+      logger.info(`Created launch record in database:  ${launchesRecord}`)
+
+      const message: MessageBody = {
+        repoName,
+        appId,
+        primaryDomainSource: primaryDomain,
+        primaryDomainTarget,
+        domainValidationSource,
+        domainValidationTarget,
+      }
+
+      if (redirectionDomainSource) {
+        message.redirectionDomain = [
+          {
+            source: redirectionDomainSource,
+            target: primaryDomainTarget,
+          },
+        ]
+      }
+
+      this.queueService.sendMessage(message)
+    } catch (error) {
+      logger.error(`Failed to created '${repoName}' site on Isomer: ${error}`)
+      throw error
+    }
+    return null
+  }
+
+  siteUpdate = async () => {
+    try {
```

Comment:
could i check why we have a `try` here?

Reply:
f7411ef

Reply:
ed0c0c0 
Right, this doesnt actually catch anything, and the more I think about this it shouldnt. The only error that would come up is when parsing the messages from the incoming queue. However, if we cannot parse it, there is no meaningful info we can append to the error message that we can log. Best effort is to log the message itself, which is done in `parseMessageBody()`